validator/src/test/java/com/amazon/aoc/validators/LoadBalanceValidatorTest.java

```diff
@@ -0,0 +1,137 @@
+    Context context = initContext();
+
+    String mockedTrace =
+        "[{ \"segments\": [{ \"document\": '{ \"metadata\": { \"default\": {\"collector-id\": 3} }, \"subsegments\": [{ \"metadata\": { \"default\": {\"collector-id\": 3} }}, { \"metadata\": { \"default\": {\"collector-id\": 3} }}] }' }] }]";
+
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    List<Trace> mockedTraceList =
+        mapper.readValue(
+            mockedTrace.getBytes(StandardCharsets.UTF_8), new TypeReference<List<Trace>>() {});
+
+    runValidation(validationConfig, context, mockedTraceList);
+  }
+
+  /** test validation when fails due to difference collector ids */
+  @Test
+  public void testValidationFailedExpectedMatchingId() throws Exception {
+    ValidationConfig validationConfig = new ValidationConfig();
+    validationConfig.setCallingType("http");
+    Context context = initContext();
+
+    String mockedTrace =
+        "[{ \"segments\": [{ \"document\": '{ \"metadata\": { \"default\": {\"collector-id\": 3} }, \"subsegments\": [{ \"metadata\": { \"default\": {\"collector-id\": 2} }}, { \"metadata\": { \"default\": {\"collector-id\": 3} }}] }' }] }]";
+
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    List<Trace> mockedTraceList =
+        mapper.readValue(
+            mockedTrace.getBytes(StandardCharsets.UTF_8), new TypeReference<List<Trace>>() {});
+
+    BaseException e =
+        assertThrows(
+            BaseException.class, () -> runValidation(validationConfig, context, mockedTraceList));
+    assertEquals(e.getCode(), ExceptionCode.COLLECTOR_ID_NOT_MATCHED.getCode());
+  }
+
+  /** test validation when it fails due to not enough spans for the test */
+  @Test
+  public void testValidationFailedExpectedMoreSpans() throws Exception {
+    ValidationConfig validationConfig = new ValidationConfig();
+    validationConfig.setCallingType("http");
+    Context context = initContext();
+
+    String mockedTrace =
+        "[{ \"segments\": [{ \"document\": '{ \"metadata\": { \"default\": {\"collector-id\": 3} }, \"subsegments\": [] }' }] }]";
+
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    List<Trace> mockedTraceList =
+        mapper.readValue(
+            mockedTrace.getBytes(StandardCharsets.UTF_8), new TypeReference<List<Trace>>() {});
+
+    BaseException e =
+        assertThrows(
+            BaseException.class, () -> runValidation(validationConfig, context, mockedTraceList));
+    assertEquals(e.getCode(), ExceptionCode.NOT_ENOUGH_SPANS.getCode());
+  }
+
+  private Context initContext() {
+    // fake vars
+    String namespace = "fakednamespace";
+    String testingId = "fakedTesingId";
+    String region = "us-west-2";
+
+    // faked context
+    Context context = new Context(testingId, region, false, true);
+    context.setMetricNamespace(namespace);
+    context.setCloudWatchContext(new CloudWatchContext());
+    context.getCloudWatchContext().setIgnoreEmptyDimSet(false);
```

Comment:
Is this actually required to be set for the trace validator tests?

Reply:
It was not necessary.  Removed now in my commit.